QueueIntegrations/Prognetics.Saga.Queue.RabbitMQ.Integration.Tests/RabbitMqSagaHostTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Polly;
+using Polly.Retry;
+using Prognetics.Saga.Queue.RabbitMQ.Configuration;
+using RabbitMQ.Client;
+using System.Text;
+using System.Text.Json;
+using Prognetics.Saga.Orchestrator.DependencyInjection;
+using Prognetics.Saga.Queue.RabbitMQ.DependecyInjection;
+using Microsoft.Extensions.Hosting;
+using Prognetics.Saga.Orchestrator.Model;
+using Prognetics.Saga.Orchestrator.DTO;
+
+namespace Prognetics.Saga.Queue.RabbitMQ.Integration.Tests;
+public sealed class RabbitMQSagaHostTests : IClassFixture<RabbitMQContainerFixture>, IDisposable
+{
+    private readonly RabbitMQContainerFixture _fixture;
+
+    private readonly RabbitMQSagaOptions _options = new ();
+    private readonly RetryPolicy<BasicGetResult?> _gettingRetryPolicy = Policy
+        .HandleResult<BasicGetResult?>(x => x == null)
+        .WaitAndRetry(10, _ => TimeSpan.FromMilliseconds(100));
+    private readonly IConnection _connection;
+    private readonly IModel _channel;
+    private readonly IBasicProperties _properties;
+
+    private readonly string _queueSource;
+    private readonly string _queueTarget;
+    private const string _exchange = "saga";
+    private readonly IServiceCollection _serviceCollection;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly SagaBackgroundService _sut;
+
+    public RabbitMQSagaHostTests(RabbitMQContainerFixture fixture)
+    {
+        _fixture = fixture;
+        _connection = _fixture.Connection;
+        _channel = _connection.CreateModel();
+        _properties = _channel.CreateBasicProperties();
+        _properties.ContentType = _options.ContentType;
+        _options.ConnectionString = fixture.Container.GetConnectionString();
+        _options.Exchange = _exchange;
+
+        _queueSource = Guid.NewGuid().ToString();
+        _queueTarget = Guid.NewGuid().ToString();
+
+        _serviceCollection = new ServiceCollection()
+            .AddLogging()
+            .AddProgneticsSaga(config => config
+                .AddModelSource<DelegateSagaModelSource>()
+                .UseRabbitMQ(_options)
+                .Services.AddSingleton(new Action<SagaModelBuilder>(builder =>
+                    builder.AddTransaction(t => t.AddStep(_queueSource, _queueTarget)))));
+
+        _serviceProvider = _serviceCollection.BuildServiceProvider();
+
+        _sut = (_serviceProvider.GetRequiredService<IHostedService>() as SagaBackgroundService)!;
+    }
+
+    [Fact]
+    public async Task WhenValidMessageWasSent_ThenAppropriateMessageShouldBeFetched()
```

Comment:
Add a comment that this test MAY fail - we need to wait for the message to reach the Q run in the container